lib/utils.bash

```diff
@@ -658,7 +658,10 @@ asdf_run_hook() {
       unset asdf_hook_fun
       ev'al' "$hook_cmd" # ignore banned command just here
     }
-    asdf_hook_fun "${@:2}"
+    asdf_hook_fun "${@:2}" || {
+      display_error "Hook \"${hook_name}\" failed."
```

Comment:
Is this consistent with other places in the codebase where hooks are invoked? I don't think we should print errors here but not in other places.

Reply:
A quick survey of the code reveals that *all* hooks appear to end up in the `asdf_run_hook` function, and as such the error message and immediate exit _is_ consistent between all types of hooks.

However, I also recognize that a hook failure (like a post-reshim-plugin hook) which returns non-zero exit code might want to be ignored.

I am starting to feel that *not* modifying the asdf_run_hook, but rather checking the exit code on 
every call would be the better way to do it.

Eg, 
`lib/commands/command-exec.bash` already validates the exit code, and passes it on if the hook fails
```
      asdf_run_hook "pre_${plugin_name}_${shim_name}" "${shim_args[@]}"
      pre_status=$?
      if [ "$pre_status" -ne 0 ]; then
        return "$pre_status"
      fi
      exec "$executable_path" "${shim_args[@]}"
```
while the exit codes are ignored for 
`lib/commands/command-install.bash`
```
        asdf_run_hook "pre_asdf_download_${plugin_name}" "$full_version"
        "${plugin_path}"/bin/download
...
        asdf_run_hook "pre_asdf_install_${plugin_name}" "$full_version"
        "${plugin_path}"/bin/install
```

I suggest reverting the change to asdf_run_hook, and looking at checking the return code handling as a separate task

Reply:
>I suggest reverting the change to asdf_run_hook, and looking at checking the return code handling as a separate task
I concur

Reply:
If you are saying change this:

```
 asdf_hook_fun "${@:2}" || {
      display_error "Hook \"${hook_name}\" failed."
```

back to this:

```
 asdf_hook_fun "${@:2}"
```

Then yes, I agree 👍

Reply:
Done